src/Components/ColorPicker/init.lua

```diff
@@ -5,314 +5,277 @@ local Constants = require(Modules.Plugin.Constants)
 				}),
-				Icon = Roact.createElement(Icon, {
-					Name = props.tagIcon,
-					LayoutOrder = 1,
+				UIPadding = Roact.createElement("UIPadding", {
+					PaddingTop = UDim.new(0, 10),
+					PaddingBottom = UDim.new(0, 10),
+					PaddingLeft = UDim.new(0, 10),
+					PaddingRight = UDim.new(0, 10),
 				}),
-				Label = Roact.createElement(TextLabel, {
-					Text = tostring(props.tagName).." - Select Color",
-					LayoutOrder = 2,
-					TextColor3 = Constants.White,
-					Font = Enum.Font.SourceSansSemibold,
-				}),
-			})
-		}),
-		Body = Roact.createElement("Frame", {
-			Size = UDim2.new(1, 0, 1, -32),
-			Position = UDim2.new(0.5, 0, 0.5, 16),
-			AnchorPoint = Vector2.new(.5, .5),
-			BackgroundColor3 = Constants.White,
-			BorderColor3 = Constants.LightGrey,
-		}, {
-			UISizeConstraint = Roact.createElement("UISizeConstraint", {
-				MaxSize = Vector2.new(500, 300),
-			}),
-			UIAspectRatioConstraint = Roact.createElement("UIAspectRatioConstraint", {
-				AspectRatio = 298 / (226 - 32), --500 / ((500-20)/2 - 4 + 32 + 20),
-			}),
-			UIPadding = Roact.createElement("UIPadding", {
-				PaddingTop = UDim.new(0, 10),
-				PaddingBottom = UDim.new(0, 10),
-				PaddingLeft = UDim.new(0, 10),
-				PaddingRight = UDim.new(0, 10),
-			}),
-			Wheel = Roact.createElement("ImageButton", {
-				Size = UDim2.new(.5, -4, 1, -32),
-				Position = UDim2.new(0, 0, 0, 0),
-				BorderColor3 = Constants.DarkGrey,
-				Image = 'rbxassetid://1357075261',
-				BackgroundColor3 = Constants.Black,
-				AutoButtonColor = false,
-				ImageTransparency = 1 - val,
-
-				[Roact.Event.MouseButton1Down] = function(rbx)
-					self:setState({
-						wheelMouseDown = true,
-					})
-				end,
-
-				[Roact.Event.InputEnded] = function(rbx, input)
-					if input.UserInputType == Enum.UserInputType.MouseButton1 and self.state.wheelMouseDown then
-						local x, y = input.Position.X, input.Position.Y
-						local pos = Vector2.new(x, y) - rbx.AbsolutePosition
-						pos = pos / rbx.AbsoluteSize
-						pos = Vector2.new(math.clamp(pos.x, 0, 1), math.clamp(pos.y, 0, 1))
-						self:setState({
-							color = Color3.fromHSV(pos.x, 1 - pos.y, val),
-							wheelMouseDown = false,
-						})
-					end
-				end,
-
-				[Roact.Event.InputChanged] = function(rbx, input)
-					if self.state.wheelMouseDown and input.UserInputType == Enum.UserInputType.MouseMovement then
-						local pos = Vector2.new(input.Position.X, input.Position.Y) - rbx.AbsolutePosition
-						pos = pos / rbx.AbsoluteSize
+				Wheel = Roact.createElement("ImageButton", {
+					Size = UDim2.new(0.5, -4, 1, 0),
+					Position = UDim2.new(0, 0, 0, 0),
+					BorderColor3 = theme:GetColor("Border"),
+					Image = 'rbxassetid://1357075261',
+					BackgroundColor3 = Constants.Black,
+					AutoButtonColor = false,
+					ImageTransparency = 1 - val,
 
+					[Roact.Event.MouseButton1Down] = function(rbx)
 						self:setState({
-							color = Color3.fromHSV(pos.x, 1 - pos.y, val),
+							wheelMouseDown = true,
```

Comment:
I called it `wheelMouseDown` because it's tracking whether the mouse is down on the hue/saturation picker specifically - `mouseButtonDown` is pretty broad for this component.